pkg/antctl/raw/traceflow/command_test.go

```diff
@@ -153,3 +229,529 @@ func TestParseFlow(t *testing.T) {
+					log.Fatal(err)
+				}
+				defer os.Remove(fakeKubeconfig.Name())
+				fakeKubeconfig.Write(tc.fakeConfigs)
+				cmd.Flags().String("kubeconfig", fakeKubeconfig.Name(), "path of kubeconfig")
+			}
+			_, _, err := getk8sClient(cmd)
+			if tc.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.EqualError(t, err, tc.expectedErr)
+			}
+		})
+	}
+}
+
+func TestGetNewTraceflow(t *testing.T) {
+	k8sClient := k8sfake.NewSimpleClientset(pod1.DeepCopyObject(), pod2.DeepCopyObject())
+	tcs := []struct {
+		name        string
+		src         string
+		dst         string
+		liveTraffic bool
+		droppedOnly bool
+		timeout     time.Duration
+		expectedTf  *v1alpha1.Traceflow
+	}{
+		{
+			name: "dummy-traceflow-dst-Pod",
+			dst:  dstPod,
+			expectedTf: &v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-dst-Pod",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Destination: v1alpha1.Destination{
+						Namespace: "default",
+						Pod:       "Pod-2",
+					},
+				},
+			},
+		},
+		{
+			name: "dummy-traceflow-src-Pod",
+			src:  srcPod,
+			expectedTf: &v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-src-Pod",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Source: v1alpha1.Source{
+						Namespace: "default",
+						Pod:       "Pod-1",
+					},
+				},
+			},
+		},
+		{
+			name: "dummy-traceflow-Pod-to-IP",
+			src:  "Pod-1",
+			dst:  IP,
+			expectedTf: &v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-Pod-to-IP",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Source: v1alpha1.Source{
+						Namespace: "default",
+						Pod:       "Pod-1",
+					},
+					Destination: v1alpha1.Destination{
+						IP: IP,
+					},
+				},
+			},
+		},
+		{
+			name:        "dummy-traceflow-IP-to-Pod-with-liveTraffic",
+			src:         IP,
+			dst:         "Pod-2",
+			liveTraffic: true,
+			expectedTf: &v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-IP-to-Pod-with-liveTraffic",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Source: v1alpha1.Source{
+						IP: IP,
+					},
+					Destination: v1alpha1.Destination{
+						Namespace: "default",
+						Pod:       "Pod-2",
+					},
+					LiveTraffic: true,
+				},
+			},
+		},
+		{
+			name: "dummy-traceflow-Pod-to-Service",
+			src:  srcPod,
+			dst:  "Service",
+			expectedTf: &v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-Pod-to-Service",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Source: v1alpha1.Source{
+						Namespace: "default",
+						Pod:       "Pod-1",
+					},
+					Destination: v1alpha1.Destination{
+						Namespace: "default",
+						Service:   "Service",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			_, localOpt := newCommandAndOptions(tc.src, tc.dst, "yaml", tc.timeout, tc.liveTraffic, tc.droppedOnly, false)
+			option = localOpt
```

Comment:
why they are needed?

Reply:
Because function getNewTraceflow uses option at various if conditional statements. So, to cover various statements of getNewTraceflow different values of option is required.